### PR TITLE
Added support for sed's `-E` alias of `-r`

### LINKED ISSUE #1277: Call sed with -E instead of -r
Even though Insights is intended to primarily run only on RHEL, I believe that every bit of portability counts. It can help running tests on a machine that‘s not supported by Insights, but runs Python and Pytest just well.

Payload retrieved from a spec is obfuscated for passwords by a _sed_ [script](https://github.com/RedHatInsights/insights-core/blob/b780ab6126645919997e86ada755fca015f0af76/insights/client/insights_spec.py#L154). Sed is running with a `-r` option to use “extended regular expression”. OpenBSD (namely macOS) version of _sed_ uses `-E` option for that. Fortunately, _GNU sed_ used on most Linuxes including RHEL and Fedora support this variant, treating this `-E` as an alias for its own `-r`. It doesn’t work the other way round though. ([source](https://unix.stackexchange.com/a/131940/190030))

I want to suggest using `-E` instead of `-r` because it is supported on more platforms and it doesn’t affect RHEL/Fedora in any way. This alias is supported from sed version 4.2, though undocumented until 4.3. RHEL6 [uses](https://access.redhat.com/downloads/content/sed/4.2.1-10.el6/x86_64/fd431d51/package) 4.2.

### PR DESCRIPTION
**Pull request by sourabhtk37**
Closes #1277 
Signed-off-by: T K Sourab <sourabhtk37@gmail.com>

### CODE REVIEW

**Comment by kylape** [comment]
ok to test

**Comment by YakovLittle** [comment]
test this please

**Comment by YakovLittle** [comment]
test this please

**Comment by eduardocerqueira** [comment]
retest this please

**Comment by esakaiev** [comment]
test this please

**Comment by kylape** [comment]
test this please

**Comment by sourabhtk37** [comment]
@kylape Anything needs to be done?

**Comment by psachin** [comment]
Hey @sourabhtk37 Can you confirm if this PR is still valid?

**Comment by sourabhtk37** [comment]
@psachin Yes, it's valid.

**Comment by sourabhtk37** [comment]
@psachin I have resolved but please give me some time to complete as I see there is few more places that needs support for the same.

**Comment by sourabhtk37** [comment]
@psachin Updated the changes.

**Comment by psachin** [comment]
@gravitypriest I think this change is valid. If you are OK with this change, can we merge it ?

**Review by psachin — APPROVED** [review]

**Review by Glutexo — CHANGES_REQUESTED** [review]

File: insights/client/insights_spec.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def get_output(self):
             return
 
         exec_start = time.time()
-        sedcmd = Popen(['sed', '', self.real_path], stdout=PIPE)
+        sedcmd = Popen(['sed', '-Ef', self.real_path], stdout=PIPE)
```
[the remark this answers is not in the document]
Comment (Glutexo) [reply insights/client/insights_spec.py:147]: The introduced bug can be even seen in the output:

```
sed: file /etc/redhat-access-insights/machine-id line 1: extra characters after command
sed: file /etc/machine-id line 1: extra characters after command
sed: file /usr/lib/udev/rules.d/40-redhat.rules line 4: unknown command: `S'
sed: file /proc/1/cgroup line 1: : doesn't want any addresses
sed: file /etc/insights-client/insights-client.conf line 1: unknown command: `['
```

**Comment by Glutexo** [comment]
I reverted the changes in _insights_spec.py_ that introduced a bug, fixing that. I verified the change in _data_collector.py_ and confirm it works fine even on RHEL 6.

**Comment by Glutexo** [comment]
@psachin May I ask you (or anyone else) for a new review?

### Steps to verify

1. Run the Insights collection.

   ```sh
   # BYPASS_GPG=True EGG=insights-core insights-client --no-gpg --offline
   Starting to collect Insights data for localhost.localdomain
   Archive saved at /var/tmp/OLvKyC/insights-localhost.localdomain-20230926145152.tar.gz
   ```

2. Inspect a file containing “password” in the archive

   ```sh
   # tar -xOf /var/tmp/OLvKyC/insights-localhost.localdomain-20230926145152.tar.gz ./insights-localhost.localdomain-20230926145152/etc/pam.d/password-auth
   …
   password    ********     ********
   …
   ```

   If the regular expression works correcly, a line containing “password” replaces all following words with “********”. A non-extended regular expression would make the replacement to not work, printing errors like
   
   > sed: file /etc/insights-client/.exp.sed line 2: invalid reference \2 on `s' command's RHS

3. Check that the output does not contain sed errors like
   
   > sed: file /etc/redhat-access-insights/machine-id line 1: extra characters after command.
   
   Although the collection would halt in case of a malfunctioning sed command in  _insights_spec.py_.

**Review by Glutexo — APPROVED** [review]

**Comment by ryan-blakley** [comment]
I just wanted to mention PR #3900 as it removes the sed command call in the data_collection.py file.

**Comment by Glutexo** [comment]
There is another pull request #3900 that removes usage of _sed_ entirely. Because of that, any changes to the sed command are unnecessary. Closing then. Thanks, @ryan-blakley!

**Pull request closed without merging**